src/Cello/disk_File.hpp

```diff
@@ -18,6 +18,9 @@ class File {
 
 public: // interface
 
+  /// factory method for constructing a FileHdf5 instance
```

Comment:
Maybe useful to explain briefly here why this is implemented as a factory method.